sdk/python/lib/pulumi_policy/serialize.py

```diff
@@ -85,3 +87,49 @@ def _deserialize_archive(prop: Dict[str, Any]) -> pulumi.Archive:
     if "uri" in prop:
         return pulumi.RemoteArchive(prop["uri"])
     raise AssertionError("Invalid archive encountered when unmarshaling resource property")
+
+def serialize_properties(props: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Serializes properties in preparation for a gRPC call.
+    """
+    result: Dict[str, Any] = {}
+    for key in props:
+        result[key] = _serialize_property(props[key])
+    return result
+
+def _serialize_property(prop: Any) -> Any:
+    # Check for proxies, and unwrap to get the raw elements. This ensures unknowns flow through.
+    if isinstance(prop, _DictProxy):
+        return _serialize_property(prop["__map"])
+    if isinstance(prop, _ListProxy):
+        return _serialize_property(prop["__elems"])
+
+    if isinstance(prop, list):
```

Comment:
We could make this more general and check for `Sequence` rather than `list`... But that has its own oddities that make it a bit more complex, where you have to explicitly check for `str` and some other things that are also `Sequence` that we don't want to treat as a `Sequence` (see https://github.com/pulumi/pulumi/blob/a935a23a41e5806c5cdeea8cf7ea4dd869947058/sdk/python/lib/pulumi/runtime/rpc.py#L330-L332), so I'd understand wanting to keep this check `isinstance(prop, list)`.